File: frontend/services/api/network.ts
import { fetchJson } from "@/services/api/client";
import { NetworkSnapshot, NetworkStats } from "@/types/api";

export async function getNetworkStats(): Promise<NetworkStats> {
  return fetchJson<NetworkStats>("/api/v1/network/stats", {
    retries: 1,
  });
}

export async function getNetworkSnapshot(): Promise<NetworkSnapshot> {
  return fetchJson<NetworkSnapshot>("/api/v1/network/snapshot", {
    retries: 1,
  });
}

File: backend/app/api/routes/network.py
from fastapi import APIRouter, Depends

from app.api.deps import get_store
from app.schemas.network import NetworkSnapshot, NetworkStats
from app.services.state_store import InMemoryStateStore

router = APIRouter(prefix="/network", tags=["network"])


@router.get("/stats", response_model=NetworkStats)
async def network_stats(store: InMemoryStateStore = Depends(get_store)) -> NetworkStats:
    return await store.network_stats()


@router.get("/snapshot", response_model=NetworkSnapshot)
async def network_snapshot(store: InMemoryStateStore = Depends(get_store)) -> NetworkSnapshot:
    return await store.network_snapshot()

